src/gui/Src/Utils/CachedFontMetrics.h

```diff
@@ -19,6 +19,14 @@ class CachedFontMetrics : public QObject
     int width(const QChar & ch)
     {
         auto unicode = ch.unicode();
+        if(unicode >= 0xD800)

```

Comment:
This is for surrogates. Do you know the details of UTF-16 encoding? Maybe you want to look at [https://en.wikipedia.org/wiki/UTF-16](https://en.wikipedia.org/wiki/UTF-16)